File: refactoring-practice-1/src/main/java/dojo/extract_interface/Employee.java
package dojo.extract_interface;

import java.util.Date;

public class Employee {

	private String firstName;
	private String lastName;
	private String middleName;
	private Date dateOfBirth;
	private double rate;
	private boolean specialSkilled;
	
	public String getFirstName() {
		return firstName;
	}
	
	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}
	
	public String getLastName() {
		return lastName;
	}
	
	public void setLastName(String lastName) {
		this.lastName = lastName;
	}
	
	public String getMiddleName() {
		return middleName;
	}
	
	public void setMiddleName(String middleName) {
		this.middleName = middleName;
	}
	
	public Date getDateOfBirth() {
		return dateOfBirth;
	}
	
	public void setDateOfBirth(Date dateOfBirth) {
		this.dateOfBirth = dateOfBirth;
	}
	
	public double getRate() {
		return rate;
	}
	
	public void setRate(double rate) {
		this.rate = rate;
	}
	
	public boolean isSpecialSkilled() {
		return specialSkilled;
	}
	
	public void setSpecialSkilled(boolean specialSkills) {
		this.specialSkilled = specialSkills;
	}
	
	
	
}

File: refactoring-practice-1/src/main/typescript/extract_interface/employee.ts
export class Employee {

    private firstName: string;
    private lastName: string;
    private middleName: string;
    private dateOfBirth: Date;
    private rate: number;
    private specialSkilled: boolean;

    public getFirstName(): string {
        return this.firstName;
    }

    public setFirstName(firstName: string) {
        this.firstName = firstName;
    }

    public getLastName(): string {
        return this.lastName;
    }

    public setLastName(lastName: string) {
        this.lastName = lastName;
    }

    public getMiddleName(): string {
        return this.middleName;
    }

    public setMiddleName(middleName: string) {
        this.middleName = middleName;
    }

    public getDateOfBirth(): Date {
        return this.dateOfBirth;
    }

    public setDateOfBirth(dateOfBirth: Date) {
        this.dateOfBirth = dateOfBirth;
    }

    public getRate(): number {
        return this.rate;
    }

    public setRate(rate: number) {
        this.rate = rate;
    }

    public isSpecialSkilled(): boolean {
        return this.specialSkilled;
    }

    public setSpecialSkilled(specialSkills: boolean) {
        this.specialSkilled = specialSkills;
    }

}
